_analyzers/token-graph.md

```diff
@@ -0,0 +1,201 @@
+---
+layout: default
+title: Token Graph
+nav_order: 150
+---
+
+# Token graphs
+
+Token graphs represent how tokens relate to each other during text analysis, especially when dealing with multi-word synonyms or compound words. They are important for accurate query matching and phrase expansion.
+
+Each token is assigned the following:
+
+- `position` – location where the token appears in the text
+
+- `positionLength` – how many positions the token spans (used in multi-word expressions)
+
+Token graphs use this information to build a graph structure of token relationships, which is later used during query parsing. Graph-aware token filters such [`synonym_graph`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/synonym-graph/) and [`word_delimiter_graph`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/word-delimiter-graph/) enable you to match phrases more accurately.
+
+Following diagram demonstrates the relationship between `position` and `positionLength` when using [`synonym_graph`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/synonym-graph/). "NYC" token is assigned `position` of `0` and `positionLength` of `3`.
+
+<img src="{{site.url}}{{site.baseurl}}/images/nyc-token-graph.png" alt="token graph" width="700">
+
+## Token graphs during indexing and querying
+
+At index time, `positionLength` is ignored and token graphs are not used.
+
+At query time, token graphs are used by queries such as:
```

Comment:
Are these the only query types that use token graphs? If so, we should say "the following queries" instead of "such as".

Reply:
no, there are additional ways the token graphs can be used, like multi_match and match_phrase_prefix. I'll re fraise this slightly now